operations/nft-multicontract.mjs

```diff
@@ -0,0 +1,72 @@
+//import { TezosToolkit } from "@taquito/taquito";
+import { MichelsonMap, TezosPreapplyFailureError } from '@taquito/taquito'
+import { char2Bytes } from '@taquito/utils'
+import { logger } from '../logger.mjs';
+// import { createRequire } from 'module'
+// const require = createRequire(import.meta.url);
+
+// const hex = require('string-hex');
+// const utf8 = require('utf8')
+
+export default async function(tezos, { gasLimit, storageLimit }) {
+	let loaded_contracts = {};
+	let load_contract = async function(contract_address) {
+		if (contract_address in loaded_contracts === false) {
+			loaded_contracts[contract_address] = await tezos.contract.at(contract_address);
+			logger.info(`token contract ${contract_address} loaded ${loaded_contracts[contract_address].parameterSchema.ExtractSignatures()}`);
+		}
+		return {
+			create_token: loaded_contracts[contract_address].methods.create_token,
+			mint_tokens: loaded_contracts[contract_address].methods.mint_tokens,
+			transfer_tokens: loaded_contracts[contract_address].methods.transfer
+		}
+	}
+
+	let op_to_transfer = async function(op) {
+		let estimate = await tezos.estimate.transfer(op.toTransferParams());
+		let params = op.toTransferParams({
+			gasLimit: Math.max(estimate.gasLimit ?? 0, gasLimit ?? 0),
+			storageLimit: Math.max(estimate.storageLimit ?? 0, storageLimit ?? 0),
+		});
+		return params
+	}
+
+	let create_token = async function(contract_address, token_id, metadata_ipfs) {
+		let contract_ops = await load_contract(contract_address);
+		if (typeof contract_ops.create_token != 'function') {
+			throw new Error("No create_token entrypoint on contract");
+		}
+		let token_info = MichelsonMap.fromLiteral({"": char2Bytes(metadata_ipfs)});
+		let create_op = contract_ops.create_token(token_id, token_info);
```

Comment:
Here, and in multiple other places, you're using `let` when you should use `const`.

I appreciate that this is probably in an attempt to be consistent with the rest of the peppermint codebase, but I think that we should aim for higher standards considering that this is a brand new file.